usaspending_api/common/sqs/sqs_work_dispatcher.py

```diff
@@ -953,19 +954,47 @@ def _set_message_visibility(self, new_visibility):
                 "Unable to set VisibilityTimeout. "
                 "Message might have previously been deleted upon completion or failure"
             )
-            log_job_message(logger=self._logger, message=message, job_type=self.worker_process_name, is_exception=True)
-            raise QueueWorkDispatcherError(message) from exc
+            log_dispatcher_message(self, message=message, is_exception=True)
+            raise QueueWorkDispatcherError(
+                message, worker_process_name=self.worker_process_name, queue_message=self._current_sqs_message
+            ) from exc
+
+
+class AbstractQueueError(Exception, metaclass=ABCMeta):
```

Comment:
nice ABC. I recommend the exception classes be stored in a dedicated file.

Reply:
I'm not opposed but is there a general good rule for this? I was thinking that they'll likely only be used (instantiated) by the `SQSWorkDispatcher`, so stay in the same module alongside it. 

Granted though, they will and can be _caught_  in lots of code throughout apps that use the dispatcher.

Reply:
Rule in this codebase or general practice?

This codebase does have a few files which exclusively contain custom exception classes. It seems straight-forward to me since they are self-contained objects and I want to encourage there usage by raising awareness of their existence. If nothing else it cleanly reduces the line count in larger files

Reply:
Done.